llarp/router/outbound_message_handler.cpp

```diff
@@ -190,34 +216,165 @@ namespace llarp
     return false;
   }
 
+  bool
+  OutboundMessageHandler::QueueOutboundMessage(const RouterID &remote,
+                                               Message &&msg,
+                                               const PathID_t &pathid)
+  {
+    MessageQueueEntry entry;
+    entry.message      = std::move(msg);
+    auto callback_copy = entry.message.second;
+    entry.router       = remote;
+    entry.pathid       = pathid;
```

Comment:
True, but it also requires the caller to know that the value will be moved and act accordingly.  Tough call imo.  Will change if desired for certain.

Reply:
I think the difference is that by taking a ref, you *always* require a copy, but by taking a value and doing a move, you sometimes incur one copy.

That said, this call is nothing compared to the network I/O that follows... :)

Reply:
I agree that it would be slightly more cpu efficient to do a move, I just don't know if it's worth it to require any part of the code calling this function to be aware of that side-effect.  I can do it either way, of course.